agent/src/main/java/com/teamscale/jacoco/agent/upload/teamscale/TeamscaleUploader.java

```diff
@@ -34,18 +30,21 @@ public TeamscaleUploader(TeamscaleServer teamscaleServer) {
 	}
 
 	@Override
-	public void upload(File coverageFile) {
+	public void upload(CoverageFile coverageFile) {
 		try (Benchmark benchmark = new Benchmark("Uploading report to Teamscale")) {
-			if (!tryUploading(coverageFile)) {
-				logger.warn("Filed to upload coverage to Teamscale. Won't delete local file");
-			} else {
+			if (tryUploading(coverageFile)) {
 				coverageFile.delete();
+			} else {
+				logger.warn("Failed to upload coverage to Teamscale. Won't delete local file {}",
+						coverageFile.getAbsolutePath());
 			}
+		} catch (IOException e) {
```

Comment:
if this is truly just for the `delete()` call, please move this around that. otherwise, something else might (in the future) raise that exception and we would log a nonsensical message then
also, please don't swallow the exception, log it
also, please tell the user what that means, i.e: "The upload was successful, but the file will be left on disk. You can delete it yourself anytime - it is no longer needed" or something similar